api/numaresourcesoperator/v1/numaresourcesoperator_defaults.go

```diff
@@ -24,11 +24,11 @@ import (
 
 func DefaultNodeGroupConfig() NodeGroupConfig {
 	ngc := NodeGroupConfig{}
-	ngc.Default()
+	ngc.FillNilFeilds()
```

Comment:
Typo: should be Fields.

But arguably we ARE filling empty resource objects with default values, aren't we? after all, "nil" is the default zero value for pointers, and we are setting it to default values - pointing to constants with default values.

In general, this field is similar to String vs ToString: wants to be like the default initialization functions into kube apis, but we're not there yet or there are small/subtle differences we want to keep, hence we use a API as similar as possible but not the real API.

Reply:
yes we can consider the nil as empty thus filling it with default. but the reason I changed this is because Default() sounds like it returns the object with default values. This is different from filling Empty/nil fields with the default because if the object was partially initialized, the non-empty fields will not be set to default. 
in this specific line it doesn't make any difference because the object is empty (was just created)

Reply:
Fair enough, so let's use `SetDefaults` which is closer (identical?) the real API does